src/bin/sage-env

```diff
@@ -370,20 +370,31 @@ if [ -n "$PYTHONHOME" ]; then
 fi
 
 if [ -n "$SAGE_LOCAL" ]; then
-    # On OS X, test whether "ld-classic" is present in the installed
-    # version of the command-line tools. If so, we add "-ld_classic"
-    # to LD_FLAGS. See #36599.
-    # When the full XCode is installed and in use, for example after
-    # "sudo xcode-select -s /Applications/Xcode.app", then "xcode-select -p"
-    # gives "/Applications/Xcode.app/Contents/Developer", but "ld-classic"
-    # is not in the subdirectory "usr/bin/" but rather in the subdirectory
-    # "Toolchains/XcodeDefault.xctoolchain/usr/bin/". See #37237.
-    # However, if LD is set explicitly, as it is within conda on macOS,
-    # do not not do this.
-    if [ "$UNAME" = "Darwin" ] && [ -z "$LD" ] && [ -x "$(xcode-select -p)/usr/bin/ld-classic" -o -x "$(xcode-select -p)/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld-classic" ] ; then
-        LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-ld_classic,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
-    else
-        LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
+    # On OS X, test whether "ld-classic" is present in the selected XCode
+    # toolchain. If so, add "-Wl,-ld_classic" to LDFLAGS (see #36599) unless
+    # LD is already set, as it will be with conda on macOS.  When the selected
+    # toolchain is in the Xcode app the output of "xcode-select -p" is
+    # "/Applications/Xcode.app/Contents/Developer", but "ld-classic" is not in
+    # the subdirectory "usr/bin/" but rather in the subdirectory
+    # "Toolchains/XcodeDefault.xctoolchain/usr/bin/".  (See #37237.)
+    LDFLAGS="$LDFLAGS -L$SAGE_LOCAL/lib"
```

Comment:
I don't think we can replace prepending in front of the user's LDFLAGS by appending. SAGE_LOCAL really needs to take precedence.

Reply:
I am not sure what you are saying.  Are you saying that we should use
 LDFLAGS="-L$SAGE_LOCAL/lib $LDFLAGS"
instead?  Do you want to search $SAGE_LOCAL/iib *before* searching a directory provided by the user?  I could do that, of course.  But it seems like the opposite of the "sage way" of doing things.  I thought sage always preferred libraries outside of its lib directory to libraries inside it.

PS I tried `@rpath` before I gave up and added the space.  It still added a link to the profile of user rpath.

Reply:
Oh yeah, I meant underscore.  I also tried space.

Reply:
> Are you saying that we should use
> LDFLAGS="-L$SAGE_LOCAL/lib $LDFLAGS"
> instead?

Yes. Just look at the code that you replaced

> Do you want to search $SAGE_LOCAL/iib _before_ searching a directory provided by the user?

Yes

>  I could do that, of course. But it seems like the opposite of the "sage way" of doing things. I thought sage always preferred libraries outside of its lib directory to libraries inside it.

No, not like this. When the Sage distribution builds stuff and installs it in SAGE_LOCAL, it needs to shadow what comes from the environment.